src/components/Contacts/ContactFlow/ContactFlow.tsx

```diff
@@ -122,14 +122,69 @@ export const ContactFlow: React.FC<Props> = ({
     enqueueSnackbar(t('Contact status info updated!'), {
       variant: 'success',
     });
-    if (status.id && taskStatuses[status.id]) {
-      openTaskModal({
-        view: TaskModalEnum.Add,
-        defaultValues: {
-          activityType: taskStatuses[status.id],
-          contactIds: [id],
-        },
-      });
+
+    switch (contactPhase) {
+      case PhaseEnum.Initiation:
+      case PhaseEnum.Appointment:
+      case PhaseEnum.FollowUp:
+      case PhaseEnum.PartnerCare:
+        await hasActiveTask({
+          variables: {
+            accountListId,
+            sortBy: TaskSortEnum.StartAtAsc,
+            first: 1,
+            tasksFilter: {
+              completed: false,
+              contactIds: [id],
+              activityType: contactPhase
+                ? getActivitiesByPhaseType(contactPhase)
+                : [],
+            },
+          },
+          onCompleted(data) {
+            const taskId = data.tasks.nodes[0]?.id || undefined;
+            if (taskId) {
+              openTaskModal({
+                view: TaskModalEnum.Complete,
+                taskId,
+              });
+            } else {
+              if (
```

Comment:
The `if` condition can be moved into the `else` block to make it an `else if` block.